packages/main/src/StepInput.ts

```diff
@@ -367,7 +367,15 @@ class StepInput extends UI5Element implements IFormElement {
 	}
 
 	get _valuePrecisioned() {
-		return this.value.toFixed(this.valuePrecision);
+		if (this.value === 0 || !this.value.toString().includes(".") || (this.value.toString().split(".")[1].length <= this.valuePrecision)) {
```

Comment:
"." might not always be the correct decimal symbol, there are locales with ",".  Example: Poland